api/repo.go

```diff
@@ -84,6 +85,41 @@ WITH RECURSIVE
                 WHERE o_type = $2);
 `
 
+const FILES_TREE_WITH_OPERATIONS = `
+-- find all ancestors of a file
+with ids as ((WITH RECURSIVE rfa AS (
+  SELECT f.*
+  FROM files f
+  WHERE f.id = %d
+  UNION
+  SELECT f.*
+  FROM files f INNER JOIN rfa ON f.id = rfa.parent_id
+) 
+SELECT id
+  FROM rfa
+  WHERE id != %d)
+
+  UNION
+
+-- find all descendants of a file
+(WITH RECURSIVE rfd AS (
+  SELECT f.*
+  FROM files f
+  WHERE f.id = %d
+  UNION
+  SELECT f.*
+  FROM files f INNER JOIN rfd ON f.parent_id = rfd.id
+) SELECT id
+  FROM rfd))
+  
+ select f.id, f.name, f.size, f.type, f.sub_type, f.mime_type, f.created_at, f.language, f.file_created_at, f.parent_id, f.published,
+ string_agg(op.type_id::text,',') as OperationIds from ids
+ join files f on f.id=ids.id
+ join files_operations fop on fop.file_id = ids.id
+ join operations op on fop.operation_id = op.id
```

Comment:
no need this join with operation table as we're only interested in operation id which we get from files_operations